linux/gatt/client.go

```diff
@@ -252,6 +252,26 @@ func (p *Client) WriteCharacteristic(c *ble.Characteristic, v []byte, noRsp bool
 	return p.ac.Write(c.ValueHandle, v)
 }
 
+// WriteLongCharacteristic writes a characteristic value that is longer than the MTU to a server. [Vol 3, Part F, 3.4.6]
+func (p *Client) WriteLongCharacteristic(c *ble.Characteristic, v []byte) error {
+	p.Lock()
+	defer p.Unlock()
+
+	offset := 0
+	prepareWriteLength := 0
+
+	for offset < len(v) {
+		prepareWriteLength = MinInt( len(v) - offset, p.conn.TxMTU()-5 )
```

Comment:
Adding a function here adds a bit of complexity when an `if` would suffice. Since `MinInt` isn't used anywhere else, I don't see that it adds much value at this time. Additionally, I'd have to go review the spec to see recall why 5 is subtracted from the MTU here. I suspect it is overhead related. Please add a const which clarifies the reason for the subtraction.


Suggestion:

```
maxLength := p.conn.TxMTU() - 5

for offset < len(v) {
    length := len(v) - offset
    if length > maxLength {
        length = maxLength
    }

    value := v[offset:offset+length]
    ...
```

Reply:
You could also avoid having to track `offset` just by advancing `v`:

```go
maxLength := p.conn.TxMTU() - 5
for len(v) > 0 {
    length := len(v)
    if length > maxLength {
        length = maxLength
    }
    ...
    v = v[length:]
}